### ISSUE TITLE
Make KoinListener work with property based testing

### ISSUE THREAD

**Issue by headsvk** [kotest/kotest#2171]
I took me some debugging to find out why my property tests with Koin fail.\
I expected `KoinListener` would reset Koin after each test, no matter if it's a basic test or a property test.

Then I found out that the second run of my simple test reuses the values saved in Koin from the first run.
```
checkAll<Boolean> { arg -> // true/false
  declareMock {
      Repository(arg)
  }
  get<Service>() // always Repository(true)
}
```

I understand I have to implement Koin support with `PropTestListener` on top of the `KoinListener` to get it to work as I expect?
Can we get `KoinPropTestListener` as a library extension?

**Comment by sksamuel** [comment]
Good idea, we can add an adapter to allow this to reset in property tests too.
